rust-code-analysis-cli/src/formats.rs

```diff
@@ -21,100 +21,118 @@ impl Format {
-                    writeln!(stdout, "{json_data}")
-                }
-                Format::Toml => {
-                    let toml_data = if pretty {
-                        toml::to_string_pretty(&space).unwrap()
-                    } else {
-                        toml::to_string(&space).unwrap()
-                    };
-                    writeln!(stdout, "{toml_data}")
-                }
-                Format::Yaml => writeln!(stdout, "{}", serde_yaml::to_string(&space).unwrap()),
-            }
+        if let Some(output_path) = output_path {
+            let format_path = self.create_unique_output_filename(path, output_path);
+            self.dump_on_file(space, format_path, pretty)
         } else {
-            let format_ext = match self {
-                Format::Cbor => ".cbor",
-                Format::Json => ".json",
-                Format::Toml => ".toml",
-                Format::Yaml => ".yml",
-            };
-
-            // Remove root /
-            let path = path.strip_prefix("/").unwrap_or(path);
-
-            // Remove root ./
-            let path = path.strip_prefix("./").unwrap_or(path);
-
-            // Replace .. with . to keep files inside the output folder
-            let cleaned_path: Vec<&str> = path
-                .iter()
-                .map(|os_str| {
-                    let s_str = os_str.to_str().unwrap();
-                    if s_str == ".." {
-                        "."
-                    } else {
-                        s_str
-                    }
-                })
-                .collect();
-
-            // Create the filename
-            let filename = cleaned_path.join("/") + format_ext;
-
-            // Build the file path
-            let format_path = output_path.as_ref().unwrap().join(filename);
-
-            // Create directories
-            create_dir_all(format_path.parent().unwrap()).unwrap();
-
-            let mut format_file = File::create(format_path)?;
-            match self {
-                Format::Cbor => serde_cbor::to_writer(format_file, &space)
-                    .map_err(|e| Error::new(ErrorKind::Other, e.to_string())),
-                Format::Json => {
-                    if pretty {
-                        serde_json::to_writer_pretty(format_file, &space)
-                            .map_err(|e| Error::new(ErrorKind::Other, e.to_string()))
-                    } else {
-                        serde_json::to_writer(format_file, &space)
-                            .map_err(|e| Error::new(ErrorKind::Other, e.to_string()))
-                    }
-                }
-                Format::Toml => {
-                    let toml_data = if pretty {
-                        toml::to_string_pretty(&space).unwrap()
-                    } else {
-                        toml::to_string(&space).unwrap()
-                    };
-                    format_file.write_all(toml_data.as_bytes())
+            self.dump_stdout(space, pretty)
```

Comment:
It's a pity that there is so much code duplication between these two functions.

Reply:
Yep, you're right, I'm going to simplify these functions in this PR